File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.cpp
class Solution {
public:
    
    int solve(string& s,int i,int sign,long res)
    {
        if(sign*res>=INT_MAX)
        {
            return INT_MAX;
        }
        
        if(res*sign<=INT_MIN)
        {
            return INT_MIN;
        }
        
        if(i>=s.length()|| s[i]<'0'|| s[i]>'9')
        {
            return sign*res;
        }
        
        res=solve(s,i+1,sign,(res*10+(s[i]-'0')));
        return res;
    }
    int myAtoi(string s) {
        
        
        int i=0,n=s.length(),sign=1;
        
        while(s[i]==' ' && i<n)
        {
            ++i;
        }
        
        if(s[i]=='-')
        {
            sign=-1;
            ++i;
        }
        else if(s[i]=='+')
        {
            ++i;
        }
        
        
       return solve(s,i,sign,0);
    }
};

File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.py
class Solution:
    def myAtoi(self, s: str) -> int:
        sign=1
        s=s.strip()
        if s=="": 
            return 0
        char=s[0]
        if char=="-" or char=="+":
            s=s[1:]
            if char=="-":
                sign=-1
        ans=0
        for ch in s:
            if '0'<=ch<='9':
                ans=ans*10+(int(ch))
            else:
                break
        ans = ans*sign     
        if ans<-2147483648:
            return -2147483648
        elif ans>2147483647:
            return 2147483647
        
        return ans

        